### PR TITLE
[IMP] [9.0] Improved_rma_tests

### PR DESCRIPTION
**Pull request by nikul-serpentcs**
Improved Customer RMA Tests.

@aheficent

### CODE REVIEW

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

**Comment by LoisRForgeFlow** [comment]
@nikul-serpentcs Please, do not forget to rebase regularly against `9.0-remove-headers` branch since I'm still pushing some  fixes.

**Comment by nikul-serpentcs** [comment]
Sure @lreficent, Noted!!

**Comment by AaronHForgeFlow** [comment]
Thank @nikul-serpentcs can you please rebase to branch 9.0-remove-headers? There are new commits on that branch

File: rma/tests/test_rma.py [hunk 1]
```diff
@@ -54,6 +54,14 @@ def setUp(self):
 
     def _create_rma_from_move(self, products2move, type, partner, dropship,
                               supplier_address_id=None):
+
+        self.picking_in = self.stockpicking.create({
```
Comment (AaronHForgeFlow) [inline rma/tests/test_rma.py]: If you are writing in self here I'd put this on setup method

File: rma/tests/test_supplier_rma.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def test_supplier_rma(self):
 
         picking.action_assign()
         picking.do_transfer()
-        for line in self.rma_supplier_id.rma_line_ids:
+        for line in self.rma_supplier_id.rma_line_ids[0]:
```
Comment (AaronHForgeFlow) [inline rma/tests/test_supplier_rma.py]: Why are you getting only the first move?

File: rma/tests/test_supplier_rma.py [hunk 3]
```diff
@@ -100,23 +100,23 @@ def test_supplier_rma(self):
                          proc.group_id])
         domain = [('group_id', 'in', list(group_ids))]
         pickings = self.stockpicking.search(domain)
-        self.assertEquals(len(pickings), 2,
+        self.assertEquals(len(pickings), 3,
```
Comment (AaronHForgeFlow) [inline rma/tests/test_supplier_rma.py:99]: It should be just one picking and 3 moves as it was before, isn't it?

File: rma/tests/test_supplier_rma.py [hunk 4]
```diff
@@ -136,22 +136,22 @@ def test_supplier_rma(self):
                                   "Wrong qty delivered")
         picking_out.action_assign()
         picking_out.do_transfer()
-        for line in self.rma_supplier_id.rma_line_ids:
-            self.assertEquals(line.qty_to_receive, 0,
+        for line in self.rma_supplier_id.rma_line_ids[0]:
```
Comment (AaronHForgeFlow) [inline rma/tests/test_supplier_rma.py:129]: So if you are checking only the first line, you know the product, so no if clause is needed. I'd iterate over all the lines as it's already implemented.

**Comment by LoisRForgeFlow** [comment]
Hi @nikul-serpentcs 

You should not have this issues rebasing as you are working in test/ folders ~with~ which aren't modified in 9.0-remove-headers.

Please follow this steps:

1. Checkout to your branch (9.0-stock-rma-tests in this case).
2. `git fetch {eficent remote}` being eficent remote the name you gave to `https://github.com/Eficent/stock-rma.git` remote.
3. `git rebase {eficent remote}/9.0-remove-headers`
4. `git push {your remote} {your branch} -f` being your remote `https://github.com/nikul-serpentcs/stock-rma.git` and your branch `9.0-stock-rma-tests` in this case.

edited: typo

**Review by AaronHForgeFlow — COMMENTED** [review]
Thanks @nikul-serpentcs Some comments below. Can you test also the quantities thats are zero in any step of the flow?

**Comment by nikul-serpentcs** [comment]
> If you are writing in self here I'd put this on setup method

@aheficent I have a used RMA move partner [here](https://github.com/nikul-serpentcs/stock-rma/blob/9.0-stock-rma-tests/rma/tests/test_rma.py#L59) .

**Comment by AaronHForgeFlow** [comment]
@nikul-serpentcs yes, but you are calling the method in the setup, https://github.com/nikul-serpentcs/stock-rma/blob/9.0-stock-rma-tests/rma/tests/test_rma.py#L51

**Comment by nikul-serpentcs** [comment]
@lreficent I did the way you mentioned, using the `git rebase`  now I think it is clean.

Actually earlier I used: `git pull {eficent remote}/9.0-remove-headers`

I think both are some but the `pull` added the new commit, right?

edited.

**Comment by nikul-serpentcs** [comment]
@aheficent Could you please review.

**Comment by LoisRForgeFlow** [comment]
@nikul-serpentcs yes, the git pull generated that ugly commit merging differences, in this case you should always rebase as you are working on top of an evolving branch.

**Comment by nikul-serpentcs** [comment]
@aheficent Could you Please take a look, I did update code.

**Comment by nikul-serpentcs** [comment]
I have a create new PR [here](https://github.com/Eficent/stock-rma/pull/16)

**Pull request closed without merging**